prototypes/entity.lua

```diff
@@ -150,18 +348,97 @@ warehouse_buffer.logistic_slots_count = 12
 local warehouse_requester = createLogisticContainer("warehouse", "requester")
 warehouse_requester.logistic_slots_count = 12
 
+function createLogisticContainerSmall(name, logistic_type)
```

Comment:
I'm not wild about having three essentially duplicate functions like this. Any chance of refactoring the original `createLogisticContainer` function to handle all three sizes? Just needs another parameter for size/type, which would be one of "basic"/"small"/"tiny".

Reply:
I honestly hated doing it how I did and wanted to just add a couple lines to your function block but again I'm a noob and didn't know how.  I'm totally down to make it cleaner and just add parameters I just need some guidance.

Reply:
Getting rid of the duplicate functions shouldn't be too hard.

Start by editing the start of `createLogisticContainer` to look something like this:

```Lua
function createLogisticContainer(name, size, logistic_type)
  local p = table.deepcopy(data.raw["container"][name.."-"..size])
  p.name = name.."-"..logistic_type.."-"..size
```

Then update uses of the function as appropriate, using `basic` for the regular ones, e.g.:

```Lua
local warehouse_requester = createLogisticContainer("warehouse", "basic", "requester")
```

The smaller versions you added would be like:

```Lua
local warehouse_buffer_small = createLogisticContainer("warehouse", "small", "buffer")
-- or
local storehouse_storage_tiny = createLogisticContainer("storehouse", "tiny", "storage")
```

I'm not sure whether it's better to add `size` before or after `logistic_type`, but you can decide which order you think is best and go with that. I don't think I really care, lol.

What little stuff I wrote isn't tested (am not even at a PC with Factorio available), but it should point you in the right direction. If you get errors or something, I can try to help more then. :smile_cat: